Adyen/Components/Base/AddressInfo.swift

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+
+/// The model for address data.
+public struct AddressInfo: Equatable, Encodable {
+
+    internal static let invalidCountry = "ZZ"
+    internal static let invalidValue = "null"
+
+    /// Create new instance of AddressInfo
+    public init(city: String? = nil,
+                country: String? = nil,
+                houseNumberOrName: String? = nil,
+                postalCode: String? = nil,
+                stateOrProvince: String? = nil,
+                street: String? = nil,
+                apartment: String? = nil) {
+        self.city = city
+        self.country = country
+        self.houseNumberOrName = houseNumberOrName
+        self.postalCode = postalCode
+        self.stateOrProvince = stateOrProvince
+        self.street = street
+        self.apartment = apartment
+    }
+
+    /// The name of the city.
+    public var city: String?
+
+    /// The two-character country code as defined in ISO-3166-1 alpha-2. For example, US.
+    /// If you don't know the country or are not collecting the country from the shopper, provide country as ZZ.
+    public var country: String?
+
+    /// The number or name of the house.
+    public var houseNumberOrName: String?
+
+    /// A maximum of five digits for an address in the US, or a maximum of ten characters for an address in all other countries.
+    public var postalCode: String?
+
+    /// State or province codes as defined in ISO 3166-2. For example, CA in the US or ON in Canada.
+    /// Required for the US and Canada.
+    public var stateOrProvince: String?
+
+    /// The name of the street.
+    /// The house number should not be included in this field; it should be separately provided via houseNumberOrName.
+    public var street: String?
+
+    /// The name or code of apartment. Optional.
+    /// Will be included into houseNumberOrName.
+    public var apartment: String?
```

Comment:
we don't need all those fields if its just postal code, right? actually we don't need a whole `AddressInfo` object, just the postal code as a string.

Reply:
we still need it for JSON mapping